tests/autoyast/autoyast_reboot.pm

```diff
@@ -20,25 +20,15 @@ use strict;
 use base 'basetest';
 use testapi;
 use utils;
+use version_utils 'is_sle';
 
 sub run {
-    # Kill ssh proactively before reboot to avoid half-open issue on zVM
-    prepare_system_shutdown;
-
-    type_string("shutdown -r now\n");
-    reset_consoles;
-
-    # We have to reconnect in next on zVM
-    if (check_var("BACKEND", "s390x")) {
-        return;
-    }
-
-    assert_screen("bios-boot",  900);
-    assert_screen("bootloader", 30);
-
-    if (check_var("BOOTFROM", "d")) {
-        assert_screen("inst-bootmenu", 60);
-    }
+    # We are already in console, so reboot from it and do not switch to x11 or root console
+    # Note, on s390x with SLE15 VNC is not running even if enabled in the profile
+    my %args = ();
+    $args{textmode}    = 1;
+    $args{keepconsole} = 1;
+    power_action('reboot', %args);
```

Comment:
Is there a specific reason why you do not use `power_action('reboot', textmode => 1, keepconsole => 1);`

Reply:
Yeah, we can put it inline. This was a must for s390x, but then it broke for normal tests as well, so I've combined these two.